voxblox/include/voxblox/integrator/tsdf_integrator.h

```diff
@@ -65,6 +69,159 @@ class TsdfIntegratorBase {
   const Config& getConfig() const { return config_; }
 
  protected:
+  // thread safe method of getting the next point in a point cloud, also does
+  // the
+  // global transform
+  class PointGetter {
+   public:
+    PointGetter(const Transformation& T_G_C, const Pointcloud& points_C,
+                const Colors& colors, const Config& config)
+        : T_G_C_(T_G_C),
+          points_C_(points_C),
+          colors_(colors),
+          config_(config),
+          origin_(T_G_C.getPosition()),
+          atomic_idx_(0) {
+      DCHECK_EQ(points_C_.size(), colors_.size());
+    }
+
+    bool getNextPoint(size_t* point_idx, Point* point_G, bool* clearing) {
+      while (atomic_idx_.load() < points_C_.size()) {
```

Comment:
If the point is too close or too far away it needs to go back and try get the next point, until it hits the end or finds a valid one